### PR TITLE
banking stage: actually aggregate tracer packet stats

### PR DESCRIPTION
**Pull request by apfitzge**
#### Problem
aggregated_tracer_packet_stats_option always started off as None in this code.

#### Summary of Changes
The if let statement was not doing anything, so just set aggregated_tracer_packet_stats_option as the result of the recv.

Fixes #

### CODE REVIEW

File: core/src/banking_stage.rs [hunk 1]
```diff
@@ -2000,19 +2000,9 @@ impl BankingStage {
         packet_count_upperbound: usize,
     ) -> Result<(Vec<PacketBatch>, Option<SigverifyTracerPacketStats>), RecvTimeoutError> {
         let start = Instant::now();
-        let mut aggregated_tracer_packet_stats_option: Option<SigverifyTracerPacketStats> = None;
-        let (mut packet_batches, new_tracer_packet_stats_option) =
```
Comment (apfitzge) [inline core/src/banking_stage.rs:2004]: we set aggregated_tracer_packet_stats_option to None here, and then get a new option back from recv_timeout.

File: core/src/banking_stage.rs [hunk 2]
```diff
@@ -2000,19 +2000,9 @@ impl BankingStage {
...
+        let (mut packet_batches, aggregated_tracer_packet_stats_option) =
             verified_receiver.recv_timeout(recv_timeout)?;
 
-        if let Some(new_tracer_packet_stats) = &new_tracer_packet_stats_option {
-            if let Some(aggregated_tracer_packet_stats) = &mut aggregated_tracer_packet_stats_option
-            {
-                aggregated_tracer_packet_stats.aggregate(new_tracer_packet_stats);
```
Comment (apfitzge) [inline core/src/banking_stage.rs:2010]: this is unreachable since `aggregated_tracer_packet_stats_option` is None

**Review by carllin — APPROVED** [review]

File: core/src/banking_stage.rs (hunk 1 above)
Comment (carllin) [reply core/src/banking_stage.rs:2004]: hmm this seems like a bad merge, should be aggregating the metrics in this loop: https://github.com/solana-labs/solana/blob/a2e7bdf50ac5e1d4c633f64f6362028b4164c003/core/src/banking_stage.rs#L2017-L2032

**Review by carllin — DISMISSED** [review]
hmm this seems like a bad merge, should be aggregating the new `_tracer_packet_stats_option` metrics read from this loop: https://github.com/solana-labs/solana/blob/a2e7bdf50ac5e1d4c633f64f6362028b4164c003/core/src/banking_stage.rs#L2017-L2032

**Comment by apfitzge** [comment]
> hmm this seems like a bad merge, should be aggregating the new `_tracer_packet_stats_option` metrics read from this loop: https://github.com/solana-labs/solana/blob/a2e7bdf50ac5e1d4c633f64f6362028b4164c003/core/src/banking_stage.rs#L2017-L2032

Yeah that seems right. This first part can still assign the option like I've done here, but it'll need to be mut, and basically move the code I deleted into the loop.

File: core/src/banking_stage.rs [hunk 3]
```diff
@@ -2000,26 +2000,26 @@ impl BankingStage {
...
-
         let mut num_packets_received: usize = packet_batches.iter().map(|batch| batch.len()).sum();
-        while let Ok((packet_batch, _tracer_packet_stats_option)) = verified_receiver.try_recv() {
+        while let Ok((packet_batch, tracer_packet_stats_option)) = verified_receiver.try_recv() {
             trace!("got more packet batches in banking stage");
             let (packets_received, packet_count_overflowed) = num_packets_received
                 .overflowing_add(packet_batch.iter().map(|batch| batch.len()).sum());
             packet_batches.extend(packet_batch);
 
+            if let Some(tracer_packet_stats) = &tracer_packet_stats_option {
```
Comment (apfitzge) [inline core/src/banking_stage.rs:2013]: moved the code that was above into here. Not sure if there's a better way to do this with options in rust, like:
"replace option OR do this"

**Review by carllin — APPROVED** [review]

File: core/src/banking_stage.rs (hunk 3 above)
Comment (carllin) [reply core/src/banking_stage.rs:2013]: This seems fine, thanks for the fix!

**Pull request merged**